components/scream/src/share/grid/remap/horizontal_remap_utility.cpp

```diff
@@ -187,10 +187,12 @@ void HorizontalMap::set_dof_gids(const view_1d<const gid_type>& dofs_gids, const
   start_timer("EAMxx::HorizontalMap::set_dof_gids");
   EKAT_REQUIRE(dofs_gids.size()>0);
   m_dofs_gids = view_1d<gid_type>("",dofs_gids.size());
+  view_1d<gid_type> l_dofs_gids("",dofs_gids.size());
   m_num_dofs = m_dofs_gids.extent(0);
   Kokkos::parallel_for("", m_num_dofs, KOKKOS_LAMBDA (const int& ii) {
-    m_dofs_gids(ii) = dofs_gids(ii)-min_dof;
+    l_dofs_gids(ii) = dofs_gids(ii)-min_dof;
   });
+  Kokkos::deep_copy(m_dofs_gids,l_dofs_gids);
```

Comment:
would something like
```
m_dof_gids = view_1d<gid_type>(l_dofs_gids,m_num_dofs);
```
be better in this context?

Reply:
You could do this:
```C++
const auto l_dofs_gids = m_dofs_gids;
// then use l_dofs_gids as you already do in the parallel_for
// then no deep_copy
```
All you need to do here is avoid the implicit `this` access. So you just need to get a local view of m_dofs_gids for the lambda to capture, not create a separate array.

Reply:
done